packages/gitbook/src/components/TableOfContents/encodeClientTableOfContents.ts

```diff
@@ -0,0 +1,113 @@
+import type { GitBookSiteContext } from '@/lib/context';
+import { getPagePaths, hasPageVisibleDescendant } from '@/lib/pages';
+import { resolveContentRef } from '@/lib/references';
+import type { ContentRef, RevisionPage } from '@gitbook/api';
+import assertNever from 'assert-never';
+
+export type ClientTOCPage = {
```

Comment:
It could be nice to do a union to distinguish `href` only on `type: 'link'`, etc